src/backends/github/API.js

```diff
@@ -67,6 +80,9 @@ export default class API {
       if (contentType && contentType.match(/json/)) {
         return this.parseJsonResponse(response);
       }
+      if (responseStatus !== 200) {
```

Comment:
A generic request method like this should probably be checking the equivalent of `response.ok`, which is "status in the range 200-299" ([response.ok on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok), see also [MDN's list of 2xx status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#Successful_responses)). Changing it to this line would fix that:

```js
      if ((responseStatus >= 200) && (responseStatus <= 299)) {
```